File: camerax_android/lib/src/legacy/core/range.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $base;

final class IntRange extends $base.Range<int> {
  final $native.IntRange obj;

  IntRange.$native(this.obj) : super.impl();

  factory IntRange({
    required int lower,
    required int upper,
  }) {
    final obj = $native.IntRange(
      lower: lower,
      upper: upper,
    );
    return IntRange.$native(obj);
  }

  @override
  Future<int> getLower() async {
    final value = await obj.getLower();
    return value;
  }

  @override
  Future<int> getUpper() async {
    final value = await obj.getUpper();
    return value;
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/Range.kt
package dev.hebei.camerax_android.legacy

import android.util.Range

class IntRange(registrar: CameraXRegistrar) : PigeonApiIntRange(registrar) {
    override fun pigeon_defaultConstructor(lower: Long, upper: Long): Stub {
        val obj = Range(lower.toInt(), upper.toInt())
        return Stub(obj)
    }

    override fun getLower(pigeon_instance: Stub): Long {
        return pigeon_instance.lower
    }

    override fun getUpper(pigeon_instance: Stub): Long {
        return pigeon_instance.upper
    }

    class Stub(internal val obj: Range<Int>) {
        val lower get() = obj.lower.toLong()
        val upper get() = obj.upper.toLong()
    }
}